### ISSUE TITLE
Version 3.1.0 is incompatible with Micronaut 4.2.x

### ISSUE THREAD

**Issue by oliverblaha** [micronaut-projects/micronaut-jms#472]
### Expected Behavior

Consistent AWS SDK should be used between Micronaut and its referenced version of micronaut-jms.

### Actual Behaviour

Different versions of the AWS SDK are used, which leads to errors:

```
[...] Caused by: java.lang.NoClassDefFoundError: software/amazon/awssdk/protocols/query/interceptor/QueryParametersToBodyInterceptor
	at software.amazon.awssdk.services.sns.DefaultSnsBaseClientBuilder.finalizeServiceConfiguration(DefaultSnsBaseClientBuilder.java:73)
	at software.amazon.awssdk.awscore.client.builder.AwsDefaultClientBuilder.finalizeChildConfiguration(AwsDefaultClientBuilder.java:185)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.syncClientConfiguration(SdkDefaultClientBuilder.java:193)
	at software.amazon.awssdk.services.sns.DefaultSnsClientBuilder.buildClient(DefaultSnsClientBuilder.java:39)
	at software.amazon.awssdk.services.sns.DefaultSnsClientBuilder.buildClient(DefaultSnsClientBuilder.java:28)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.build(SdkDefaultClientBuilder.java:157)
	at io.micronaut.aws.sdk.v2.service.AwsClientFactory.syncClient(AwsClientFactory.java:110)
	at io.micronaut.aws.sdk.v2.service.sns.SnsClientFactory.syncClient(SnsClientFactory.java:80)
	at io.micronaut.aws.sdk.v2.service.sns.$SnsClientFactory$SyncClient1$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
	... 124 more
```

### Version

4.2.3

**Comment by sdelamo** [comment]
let me check @oliverblaha do you have a sample app?

**Comment by sdelamo** [comment]
Please note https://github.com/micronaut-projects/micronaut-jms/releases/tag/v3.1.0 will be added to the Micronaut Framework 4.3.0 BOM which is not yet released. 

Micronaut Framework 4.2.3 works with 3.0.1.

**Comment by oliverblaha** [comment]
@sdelamo It looks like Micronaut 4.2.x already uses 3.1.0, at least that's what it pulls by default. Micronaut 4.1.x indeed still uses 3.0.1.

**Comment by sdelamo** [comment]
my bad @oliverblaha you are right. let me understand what's going on.

**Comment by sdelamo** [comment]
@oliverblaha forcing the `software.amazon.awssdk:sqs` dependency to `2.21.15` should work. For example, with gradle:

[source,kotlin]
----
configurations.all {
    resolutionStrategy {
        force("software.amazon.awssdk:sqs:2.21.15")
    }
}
----

**Comment by oliverblaha** [comment]
Thanks @sdelamo, the workaround isn't really working when using renovate though. We'll probably just be waiting for the 4.3.0 release instead. But maybe there's a way to catch this in some test case before it happens again?

**Comment by oliverblaha** [comment]
btw, seems it was already reported earlier, but remained unnoticed: https://github.com/micronaut-projects/micronaut-platform/issues/1133

**Comment by sdelamo** [comment]
> Thanks @sdelamo, the workaround isn't really working when using renovate though. We'll probably just be waiting for the 4.3.0 release instead. But maybe there's a way to catch this in some test case before it happens again?

what do you mean by `when using renovate`?

**Comment by oliverblaha** [comment]
We're using renovate to auto-update dependencies; manually forcing a version for AWS defeats the purpose of automatic updates.

**Comment by sdelamo** [comment]
I created a demo application with 4.2.3 and Micronaut JMS SQS, and I am not able to replicate the error @oliverblaha 

https://github.com/sdelamo/demo-micronaut-sqs

Do you know what I am missing?

**Comment by oliverblaha** [comment]
@sdelamo This particular conflict shows when injecting SnsClient. I've raised a PR with a minimal example for your demo project.

**Comment by sdelamo** [comment]
@oliverblaha thanks for reporting and helping me work through this issue. This is fixed in Micronaut Framework 4.2.4.
